xdsl/dialects/csl.py

```diff
@@ -1511,6 +1571,20 @@ def verify_(self) -> None:
                 "Input and output type must be of different signedness"
             )
 
+    @staticmethod
+    def get(op: SSAValue | Operation) -> SignednessCastOp:
+        typ = op.results[0].type if isinstance(op, Operation) else op.type
+        assert isinstance(typ, IntegerType)
+        return SignednessCastOp(operands=[op], result_types=[IntegerType(typ.width)])
+
+    @staticmethod
+    def get_u(op: SSAValue | Operation) -> SignednessCastOp:
+        typ = op.results[0].type if isinstance(op, Operation) else op.type
+        assert isinstance(typ, IntegerType)
+        return SignednessCastOp(
+            operands=[op], result_types=[IntegerType(typ.width, Signedness.UNSIGNED)]
+        )
```

Comment:
I would prefer to see these two as a single `def __init__(self, inp: SSAValue | Operation, result_type: IntegerType)`

Reply:
actually, we don't need the result type if the behaviour is simply to flip between unsigned and signless (since we don't want explicit 'signed')

Reply:
otherwise done as suggested